### ISSUE TITLE
Error in Phantomjs

### ISSUE THREAD

**Issue by riophae** [KenanY/create-event#35]
Run this code with PhantomJS:

```js
// test.js
var event = document.createEvent('KeyboardEvent')
event.initKeyboardEvent('input')
Object.defineProperty(event, 'keyCode', {
  get: function() {
    return 13
  }
})
```

Error message:
```
TypeError: Attempting to change access mechanism for an unconfigurable property.

  phantomjs://code/test.js:3 in defineProperty
```

**Comment by riophae** [comment]
Hey, seems this issue has been ignored?

**Comment by KenanY** [comment]
@riophae Oh sorry, I was under the impression that you were working on https://github.com/KenanY/create-event/pull/36 as a fix for this :)

**Comment by riophae** [comment]
@KenanY Yeah, I was about to do that. But need some help!

**Comment by KenanY** [comment]
Seems like [`charCode`](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/charCode), [`keyCode`](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode), and [`which`](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/which) are all deprecated now (and may have been for a long time), so I think we should simply try and remove them completely, if possible.

**Comment by riophae** [comment]
Yeah, they are indeed all deprecated and the recommended one is `event.key` according to [MDN](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent). However, [`event.key` seems not having a wide range of well support](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key).

**Comment by KenanY** [comment]
Hmm that sucks 😕

**Comment by riophae** [comment]
I believe the most users of create-event are the component lib authors who would like their component to have as more browsers supported as possible. So the compatibility issue really matters.

Currently [the best choice is `event.which`](https://css-tricks.com/snippets/javascript/javascript-keycodes/) which is the best supported one and works fine with all keys. However, it's deprecated. The lib authors should handle the case where `event.which` is not present.

As for `event.key`, it should be adopted whenever possible since it's the recommended one. However, it's not as reliable as `event.which` currently - IE has a non-standard implementation of it. e.g., the space key has different results on IE11 and Chrome:

![capture](https://user-images.githubusercontent.com/1289884/29949246-f671d424-8ee5-11e7-8ef8-50b8ba2cd1df.PNG)
![default](https://user-images.githubusercontent.com/1289884/29949247-f672e512-8ee5-11e7-848b-153ff2ab5a03.PNG)

With these facts in mind, `event.which` is still the best choice for now - which means the issue we have with `event.key` in create-event is not that bad. We can just assume the lib authors are relying on `event.which` more and are able to tolerate the absence of `event.key`.

Also, we should prepare for the migration to `event.key`. We should push PhantomJS to fix the issue.

**Comment by riophae** [comment]
Oh, I am sorry... I just saw the code again and found that our issue is not only with `event.key` but with all these properties?
I'm puzzled again :(
